src/contract-logic/price-feed/index.ts

```diff
@@ -0,0 +1,40 @@
+import { Decimal } from 'decimal.js';
+import { Market } from '../../subgraph';
+
+// Converts the amount of market tokens to collateral tokens considering the decimal
+// values of both market and collateral tokens
+// Normalized price is the price of market and collateral with 8 decimals
+// The value returned by the function contains `collateralDecimals` number of decimal digits
+export const convertMarketAmountToCollateral = (
+  marketAmount: Decimal,
+  marketDecimals: Decimal,
+  collateralDecimals: Decimal,
+  normalizedMarketPrice: Decimal,
+  normalizedCollateralPrice: Decimal,
+): Decimal => {
+  const numeratorMultiplier = new Decimal('10').pow(collateralDecimals);
+  const denominatorMultiplier = new Decimal('10').pow(marketDecimals);
+
+  let tokenAmount = normalizedMarketPrice.mul(marketAmount).mul(numeratorMultiplier);
+  return tokenAmount.div(normalizedCollateralPrice).div(denominatorMultiplier);
+};
+
+// Converts marketAmount to USD with 8 decimals
+export const convertMarketAmountToUsd = (
+  marketAmount: Decimal,
+  marketDecimals: Decimal,
+  normalizedMarketPrice: Decimal,
+): Decimal => {
+  const denominatorMultiplier = new Decimal('10').pow(marketDecimals);
+  return marketAmount.mul(normalizedMarketPrice).div(denominatorMultiplier);
+};
+
+// Converts the amount of collateral to USD with 8 decimals
+export const convertCollateralAmountToUsd = (
```

Comment:
The reason why we have it as a separate function is so that its independent of the price. So the price can also be passed from the interface directly and we don't have to query the Pyth endpoints for this.
I'll add separate single function that achieves the logic by fetching prices from Pyth.